### PR TITLE
feat!: update the return value for `n=1` in `math/base/special/bernoulli`

### PR DESCRIPTION
**Pull request by gunjjoshi**
## Description

> What is the purpose of this pull request?

This pull request:

-   addresses the discussion at https://github.com/stdlib-js/stdlib/pull/3037#issuecomment-2468411692
-   adds a conditional statement to check if `n = 1`, and return `0.5` in [`math/base/special/bernoulli`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/bernoulli).
-   adds the corresponding test.
-   modifies the examples accordingly.
-   updates benchmarks and tests according to the latest conventions.

## Related Issues

> Does this pull request have any related issues?

None.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/bernoulli/benchmark/c/native/benchmark.c [hunk 1]
```diff
@@ -91,15 +91,18 @@ static double rand_double( void ) {
...
 
+	for ( i = 0; i < 100; i++ ) {
+		x[ i ] = ( 500.0 * rand_double() );
+	}
+
 	t = tic();
 	for ( i = 0; i < ITERATIONS; i++ ) {
-		x = ( 500.0 * rand_double() );
-		y = stdlib_base_bernoulli( x );
+		y = stdlib_base_bernoulli( x[ i % 100 ] );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/bernoulli/benchmark/c/native/benchmark.c]:
```suggestion
		y = stdlib_base_bernoulli( (int)( x[ i % 100 ] ) );
```

Making the input argument cast more explicit.

File: lib/node_modules/@stdlib/math/base/special/bernoulli/src/main.c [hunk 2]
```diff
@@ -173,6 +173,9 @@ double stdlib_base_bernoulli( const int32_t n ) {
 	if ( stdlib_base_is_nan( n ) || !stdlib_base_is_nonnegative_integer( n ) ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/bernoulli/src/main.c]: @gunjjoshi We should rethink these checks. 1. `n` can never be `NaN`. 2. `stdlib_base_is_nonnegative_integer` assumes a `double` as an input argument. Here we can just use `n < 0`.

File: lib/node_modules/@stdlib/math/base/special/bernoulli/test/test.js [hunk 3]
```diff
@@ -44,31 +44,37 @@ tape( 'if provided a negative number, the function returns `NaN`', function test
...
 		v = bernoulli( i );
-		t.strictEqual( isnan( v ), true, 'returns NaN when provided ' + i );
+		t.strictEqual( isnan( v ), true, 'returns expected value when provided ' + i );
 	}
 	t.end();
 });
 
+tape( 'if provided `1`, the function returns `0.5`', function test( t ) {
+	var v = bernoulli( 1 );
+	t.strictEqual( v, 0.5, 'returns expected value when provided 1' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/bernoulli/test/test.js]:
```suggestion
	t.strictEqual( v, 0.5, 'returns expected value' );
```

File: lib/node_modules/@stdlib/math/base/special/bernoulli/test/test.js [hunk 4]
```diff
@@ -44,31 +44,37 @@ tape( 'if provided a negative number, the function returns `NaN`', function test
...
+	t.end();
+});
+
 tape( 'if provided `NaN`, the function returns `NaN`', function test( t ) {
 	var v = bernoulli( NaN );
-	t.strictEqual( isnan( v ), true, 'returns NaN when provided a NaN' );
+	t.strictEqual( isnan( v ), true, 'returns expected value when provided a NaN' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/bernoulli/test/test.js]:
```suggestion
	t.strictEqual( isnan( v ), true, 'returns expected value' );
```

File: lib/node_modules/@stdlib/math/base/special/bernoulli/test/test.native.js [hunk 5]
```diff
@@ -53,19 +53,25 @@ tape( 'if provided a negative number, the function returns `NaN`', opts, functio
...
 		v = bernoulli( i );
-		t.strictEqual( isnan( v ), true, 'returns NaN when provided ' + i );
+		t.strictEqual( isnan( v ), true, 'returns expected value when provided ' + i );
 	}
 	t.end();
 });
 
+tape( 'if provided `1`, the function returns `0.5`', function test( t ) {
+	var v = bernoulli( 1 );
+	t.strictEqual( v, 0.5, 'returns expected value when provided 1' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/bernoulli/test/test.native.js]:
```suggestion
	t.strictEqual( v, 0.5, 'returns expected value' );
```

**Comment by kgryte** [comment]
Note that this change is not a "refactor", per se, but a breaking change. Upon merging, we will need to articulate that breaking change to ensure that the changelog is populated correctly.

File: lib/node_modules/@stdlib/math/base/special/bernoulli/src/main.c (hunk 2 above)
Comment (gunjjoshi) [reply lib/node_modules/@stdlib/math/base/special/bernoulli/src/main.c]: Yes, that does make sense. Also, we have already excluded the test in which `NaN` is given as the input for `test.native.js`, while we have that in `test.js`. I have made these changes, thanks!

**Comment by gunjjoshi** [comment]
> Note that this change is not a "refactor", per se, but a breaking change. Upon merging, we will need to articulate that breaking change to ensure that the changelog is populated correctly.

Got it, thanks for explaining!

File: lib/node_modules/@stdlib/math/base/special/bernoulli/test/test.native.js [hunk 6]
```diff
@@ -53,19 +53,25 @@ tape( 'if provided a negative number, the function returns `NaN`', opts, functio
...
 
 	for ( i = -1; i > -100; i-- ) {
 		v = bernoulli( i );
-		t.strictEqual( isnan( v ), true, 'returns NaN when provided ' + i );
+		t.strictEqual( isnan( v ), true, 'returns expected value when provided ' + i );
 	}
 	t.end();
 });
 
+tape( 'if provided `1`, the function returns `0.5`', function test( t ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/bernoulli/test/test.native.js]:
```suggestion
tape( 'if provided `1`, the function returns `0.5`', opts, function test( t ) {
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @gunjjoshi!

**Pull request merged**